tests/integration/Tribe/Events/iCalTest.php

```diff
@@ -0,0 +1,189 @@
+		$sut = $this->make_instance();
+
+		$this->assertInstanceOf( iCal::class, $sut );
+	}
+
+	public function count_scenarios() {
+		return [
+			// events total count, posts per page, default export count, expected
+			[ 1, 10, 15, 1 ],
+			[ 5, 10, 15, 5 ],
+			[ 12, 10, 15, 12 ],
+			[ 15, 10, 15, 15 ],
+			[ 17, 10, 15, 15 ],
+			[ 17, 20, 15, 15 ],
+			[ 13, 20, 15, 13 ],
+			[ 13, 20, 11, 11 ],
+			[ 13, 1, 11, 11 ],
+			[ 13, - 1, 7, 7 ],
+			[ 13, 15, 1, 1 ],
+		];
+	}
+
+	/**
+	 * @test
+	 * it should export the expected number of events in respect to default count, events count and posts_per_page
+	 * @dataProvider count_scenarios
+	 */
+	public function it_should_export_the_expected_number_of_events_in_respect_to_default_value_events_count_and_posts_per_page(
+		$events_total_count,
+		$posts_per_page,
+		$default_export_count,
+		$expected
+	) {
+		$post_type = \Tribe__Events__Main::POSTTYPE;
+
+		for ( $i = 0; $i < $events_total_count; $i ++ ) {
+			$meta = [ '_EventStartDate' => date( \Tribe__Date_Utils::DBDATETIMEFORMAT, strtotime( '+' . ( $i + 1 ) . ' days' ) ) ];
+			$this->factory()->post->create( [ 'post_type' => $post_type, 'meta_input' => $meta ] );
+		}
+
+		/** @var \WP_Query $wp_query */
+		global $wp_query;
+
+		// run the query, simulates what would happen on the page
+		$wp_query = new \WP_Query( [ 'post_type' => $post_type, 'posts_per_page' => $posts_per_page ] );
+
+		$sut = $this->make_instance();
+		$sut->set_feed_default_export_count( $default_export_count );
+		$content = $sut->generate_ical_feed( null, false );
+
+		$this->assertEventsCount( $expected, $content );
+	}
+
+	public function count_scenarios_with_filter() {
+		return [
+			// events total count, posts per page, default export count, filter count, expected
+			[ 1, 10, 15, 5, 1 ],
+			[ 5, 10, 15, 5, 5 ],
+			[ 12, 10, 15, 11, 11 ],
+			[ 15, 10, 15, 15, 15 ],
+			[ 17, 10, 15, 10, 10 ],
+		];
+	}
+
+	/**
+	 * @test
+	 * it should export the expected number of events in respect to default count, events count, posts_per_page and filter value
+	 * @dataProvider count_scenarios_with_filter
+	 */
+	public function it_should_export_the_expected_number_of_events_in_respect_to_context_and_filter(
+		$events_total_count,
+		$posts_per_page,
+		$default_export_count,
+		$filter_count,
+		$expected
+	) {
+		$post_type = \Tribe__Events__Main::POSTTYPE;
+
+		for ( $i = 0; $i < $events_total_count; $i ++ ) {
+			$meta = [ '_EventStartDate' => date( \Tribe__Date_Utils::DBDATETIMEFORMAT, strtotime( '+' . ( $i + 1 ) . ' days' ) ) ];
+			$this->factory()->post->create( [ 'post_type' => $post_type, 'meta_input' => $meta ] );
+		}
+
+		/** @var \WP_Query $wp_query */
+		global $wp_query;
+
+		// run the query, simulates what would happen on the page
+		$wp_query = new \WP_Query( [ 'post_type' => $post_type, 'posts_per_page' => $posts_per_page ] );
+
+		add_filter(
+			'tribe_ical_feed_posts_per_page', function () use ( $filter_count ) {
+			return $filter_count;
+		}
```

Comment:
Spacing/indentation is weird here. How about:

```
add_filter(
	'tribe_ical_feed_posts_per_page', function () use ( $filter_count ) {
		return $filter_count;
	}
);
```

(Or even have the closing function brace and parethensis on the same line, like `} );`?